File: src/main/java/Spring/Entities/Bands.java
package Spring.Entities;

import javax.persistence.*;


@Entity
@Table(name = "Bands", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"id"})
})
public class Bands {

    @GeneratedValue
    @Id
    @Column(name = "id")
    private int id;

    @Column(name = "Band_Name" , unique = true)
    private String bandName;

    @Column(name = "Year_of_creation")
    private int yearOfCreation;

    @Column(name = "Origin")
    private String origin;

    @Column(name = "Average_Rate_Of_Band")
    private float averageRateOfBand;

    public Bands() {
    }

    public Bands(String bandName, int yearOfCreation, String origin, float averangeRateOfBand) {
        this.bandName = bandName;
        this.yearOfCreation = yearOfCreation;
        this.origin = origin;
        this.averageRateOfBand = averangeRateOfBand;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getBandName() {
        return bandName;
    }

    public void setName(String bandName) {
        this.bandName = bandName;
    }

    public int getYearOfCreation() {
        return yearOfCreation;
    }

    public void setYearOfCreation(int yearOfCreation) {
        this.yearOfCreation = yearOfCreation;
    }

    public String getOrigin() {
        return origin;
    }

    public void setOrigin(String origin) {
        this.origin = origin;
    }

    public float getAverangeRateOfBand() {
        return averageRateOfBand;
    }

    public void setAverangeRateOfBand(float averangeRateOfBand) {
        this.averageRateOfBand = averangeRateOfBand;
    }


}

File: FrontEnd/src/app/Class/bands.ts
export class Bands{

	private id: number;
	private bandName: String;
	private yearOfCreation: number;
	private origin: String;
	private averageRateOfBand: number;
	
	constructor(obj?: any){
		
		this.id = (obj && obj.id) || 0;
		this.bandName = (obj && obj.bandName) || '';
		this.yearOfCreation = (obj && obj.yearOfCreation) || 0;
		this.origin = (obj && obj.origin) || '';
		this.averageRateOfBand = (obj && obj.averageRateOfBand) || 0;
	
	}
	
	getId(): number{
		return this.id;
	}
	setId(id: number){
		this.id = id;
	}
	
	getBandName(): String{
		return this.bandName;
	}
	setBandName(bandName: String){
		this.bandName = bandName;
	}
	
	getYearOfCreation(): number{
		return this.yearOfCreation;
	}
	setYearOfCreation(yearOfCreation: number){
		this.yearOfCreation = yearOfCreation;
	}
	
	getOrigin(): String{
		return this.origin;
	}
	setOrigin(origin: String){
		this.origin = origin;
	}
	
	getAverageRateOfBand(){
		return this.averageRateOfBand;
	}
	setAverageRateOfBand(averageRateOfBand: number){
		this.averageRateOfBand = averageRateOfBand;
		
	}

}